src/iio_wrapper.cpp

```diff
@@ -693,3 +729,139 @@ int32_t IIOWrapper::decimation_filter()
+}
+
+int IIOWrapper::update_burst_data_selection(int32_t val)
+{
+    long long valuel = val;
+    return iio_device_debug_attr_write_longlong(m_dev,"burst_data_selection", valuel);
+}
+
+int IIOWrapper::update_linear_acceleration_compensation(int32_t val)
+{
+    long long valuel = val;
+    return iio_device_debug_attr_write_longlong(m_dev,"linear_acceleration_compensation", valuel);
+}
+
+int IIOWrapper::update_point_of_percussion_alignment(int32_t val)
+{
+    long long valuel = val;
+    return iio_device_debug_attr_write_longlong(m_dev,"point_of_percussion_alignment", valuel);
+}
+
+int IIOWrapper::update_internal_sensor_bandwidth(int32_t val)
+{
+    long long valuel = val;
+    return iio_device_debug_attr_write_longlong(m_dev,"internal_sensor_bandwidth", valuel);
+}
+
+int IIOWrapper::update_sync_mode_select(int32_t val)
+{
+    long long valuel = val;
+    return iio_device_debug_attr_write_longlong(m_dev,"sync_mode_select", valuel);
+}
+
+int IIOWrapper::update_sync_polarity(int32_t val)
+{
+    long long valuel = val;
+    return iio_device_debug_attr_write_longlong(m_dev,"sync_polarity", valuel);
+}
+
+int IIOWrapper::update_data_ready_polarity(int32_t val)
+{
+    long long valuel = val;
+    return iio_device_debug_attr_write_longlong(m_dev,"data_ready_polarity", valuel);
+}
+
+int IIOWrapper::update_sync_signal_scale(int32_t val)
+{
+    long long valuel = val;
+    return iio_device_debug_attr_write_longlong(m_dev,"sync_signal_scale", valuel);
+}
+
+int IIOWrapper::update_decimation_filter(int32_t val)
+{
+    long long valuel = val;
+    return iio_device_debug_attr_write_longlong(m_dev,"decimation_filter", valuel);
+}
+
+int IIOWrapper::update_accel_calibbias_x(int32_t val)
+{
+    long long valuel = val;
+    return iio_channel_attr_write_longlong(m_channel_accel_x, "calibbias", valuel);
+}
+
+int IIOWrapper::update_accel_calibbias_y(int32_t val)
+{
+    long long valuel = val;
+    return iio_channel_attr_write_longlong(m_channel_accel_y, "calibbias", valuel);
+}
+
+int IIOWrapper::update_accel_calibbias_z(int32_t val)
+{
+    long long valuel = val;
+    return iio_channel_attr_write_longlong(m_channel_accel_z, "calibbias", valuel);
+}
+
+int IIOWrapper::update_anglvel_calibbias_x(int32_t val)
+{
+    long long valuel = val;
+    return iio_channel_attr_write_longlong(m_channel_anglvel_x, "calibbias", valuel);
+}
+
+int IIOWrapper::update_anglvel_calibbias_y(int32_t val)
+{
+    long long valuel = val;
+    return iio_channel_attr_write_longlong(m_channel_anglvel_y, "calibbias", valuel);
+}
+
+int IIOWrapper::update_anglvel_calibbias_z(int32_t val)
+{
+    long long valuel = val;
+    return iio_channel_attr_write_longlong(m_channel_anglvel_z, "calibbias", valuel);
+}
+
+int IIOWrapper::update_sampling_frequency(int32_t val)
+{
+    long long valuel = val;
+    return iio_device_attr_write_longlong(m_dev, "sampling_frequency", valuel);
+}
```

Comment:
the update function should work like this:
read current set value
compare current set value with the desired value
if the two values are different write the desired value